### PR TITLE
23 clustering

### PR DESCRIPTION
**Pull request by annprzy**

### CODE REVIEW

File: src/pipeline/transforms.py [hunk 1]
```diff
@@ -1,18 +1,19 @@
-import os
 from abc import ABC, abstractmethod
 
 from typing import Dict, Union, Callable
 
+from typing import Tuple
```
Comment (jkarolczak) [inline src/pipeline/transforms.py:5]: I would merge this section into single:
```python
from typing import Callable, Dict, Tuple, Union
```

File: tests/test_pipeline.py [hunk 2]
```diff
@@ -48,3 +48,11 @@ def test_random_selection(config):
...
         assert np.sum(transformed > 0) <= np.sum(transformed > 0)
 
+
+@pytest.mark.parametrize("config", [{"max_blob_size": 200}, {"max_blob_size": 1000}, {"max_blob_size": 2000}])
+def test_clustering(config):
+    transforms = ClusteringTransform(config)
+    for blob in blobs():
+        transformed = transforms.preprocess(blob)
+        assert transformed.shape == blob.shape
+        assert np.sum(transformed > 0) <= np.sum(transformed > 0)
```
Comment (jkarolczak) [inline tests/test_pipeline.py:58]: It should be
```python
if np.sum(blob > 0) > config["max_blob_size"]:
    assert np.sum(transformed > 0) == config["max_blob_size"]
else:
    assert np.sum(transformed > 0) == np.sum(blob > 0)
```

I know that there is a similar issue related to the test of shell extraction, I will fix it soon

**Review by jkarolczak — CHANGES_REQUESTED** [review]
sorry

**Comment by jkarolczak** [comment]
@annprzy I suggested some improvements and fixes on a branch [23-clustering-suggestions](https://github.com/jkarolczak/ligand-classification/tree/23-clustering-suggestions). Please consider merging it to this PR, then I will gladly change my change request to approve

**Review by wtaisner — APPROVED** [review]

File: src/pipeline/transforms.py (hunk 1 above)
Comment (jkarolczak) [reply src/pipeline/transforms.py]: ✅ fixed

**Review by jkarolczak — APPROVED** [review]

**Comment by jkarolczak** [comment]
@annprzy  after all
well done

**Pull request merged**